reactive/aws.py

```diff
@@ -6,72 +6,60 @@
 
-from charms.layer.aws import (
-    update_credentials_file,
-    tag_instance,
-    tag_instance_security_group,
-    tag_instance_subnet,
-    enable_elb,
-    enable_ebs,
-    enable_route53,
-    enable_s3_read,
-    enable_s3_write,
-)
+from charms.layer import aws as charm_lib
+from charms.layer import status
 
 
-@when_any('config.changed.access_key',
-          'config.changed.secret_key')
-def update_credentials():
-    config = hookenv.config()
-    access_key = config['access_key']
-    secret_key = config['secret_key']
-    update_credentials_file(access_key, secret_key)
-    toggle_flag('charm.aws.has_secrets',
-                access_key and secret_key)
+@when_any('config.changed.access-key',
+          'config.changed.secret-key')
+def update_creds():
+    clear_flag('charm.aws.creds.set')
 
 
-@when_not('charm.aws.has_secrets')
-def no_secrets():
-    hookenv.status_set('blocked', 'cloud credential access required')
+@when_not('charm.aws.creds.set')
+def get_creds():
+    toggle_flag('charm.aws.creds.set', charm_lib.get_credentials())
 
 
 @when_all('snap.installed.aws-cli',
-          'charm.aws.has_secrets')
+          'charm.aws.creds.set')
 @when_not('endpoint.aws.requested')
 def no_requests():
-    hookenv.status_set('active', 'ready')
+    status.active('ready')
 
 
 @when_all('snap.installed.aws-cli',
-          'charm.aws.has_secrets',
+          'charm.aws.creds.set',
           'endpoint.aws.requested')
 def handle_requests():
-    hookenv.status_set('maintenance', 'granting integration requests')
+    status.maintenance('granting integration requests')
     aws = endpoint_from_flag('endpoint.aws.requested')
     for request in aws.requests:
         if request.instance_tags:
-            tag_instance(request.instance_id,
-                         request.region,
-                         request.instance_tags)
+            charm_lib.tag_instance(
+                request.instance_id,
+                request.region,
+                request.instance_tags)
         if request.instance_security_group_tags:
-            tag_instance_security_group(request.instance_id,
-                                        request.region,
-                                        request.instance_security_group_tags)
+            charm_lib.tag_instance_security_group(
+                request.instance_id,
+                request.region,
+                request.instance_security_group_tags)
         if request.instance_subnet_tags:
-            tag_instance_subnet(request.instance_id,
-                                request.region,
-                                request.instance_subnet_tags)
+            charm_lib.tag_instance_subnet(
+                request.instance_id,
+                request.region,
+                request.instance_subnet_tags)
         if request.requested_elb:
-            enable_elb(request.instance_id, request.region)
+            charm_lib.enable_elb(request.instance_id, request.region)
```

Comment:
Dont forget to update these to deal with the new method sigs from #4 (needs an application name as the 1st param).